lib/include/jxl/encode.h

```diff
@@ -264,8 +264,123 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderAddImageFrame(
     const JxlEncoderOptions* options, const JxlPixelFormat* pixel_format,
     const void* buffer, size_t size);
 
+/** Adds a metadata box to the file format. JxlEncoderProcessOutput must be used
+ * to effectively write the box to the output. @ref JxlEncoderUseContainer must
+ * be enabled before using this function.
+ *
+ * Background information about the container format and boxes follows here:
+ *
+ * For users of libjxl, boxes allow inserting application-specific data and
+ * metadata (Exif, XML, JUMBF and user defined boxes).
+ *
+ * The box format follows ISO BMFF and shares features and box types with other
+ * image and video formats, including the Exif, XML and JUMBF boxes. The box
+ * format for JPEG XL is specified in ISO/IEC 18181-2.
+ *
+ * Boxes in general don't contain other boxes inside, except a JUMBF superbox.
+ * Boxes follow each other sequentially and are byte-aligned. If the container
+ * format is used, the JXL stream exists out of 3 or more concatenated boxes.
+ * It is also possible to use a direct codestream without boxes, but in that
+ * case metadata cannot be added.
+ *
+ * Each box generally has the following byte structure in the file:
+ * - 4 bytes: box size including box header (Big endian. If set to 0, an
+ *   8-byte 64-bit size follows instead).
+ * - 4 bytes: type, e.g. "JXL " for the signature box, "jxlc" for a codestream
+ *   box.
+ * - N bytes: box contents.
+ *
+ * Only the box contents are provided to the contents argument of this function,
+ * the encoder encodes the size header itself.
+ *
+ * Box types are given by 4 characters. A list of known types follows:
+ * - "JXL ": mandatory signature box, must come first, 12 bytes long including
+ *   the box header
+ * - "ftyp": a second mandatory signature box, must come second, 20 bytes long
```

Comment:
Theoretically it could be > 20 bytes long too.

Reply:
The specification currently specifies those exact 20 bytes, so for now this makes sense I'd say